modules/service/controllers/globalrbac/handler.go

```diff
@@ -43,9 +47,91 @@ func Register(ctx context.Context, rContext *types.Context) error {
 	rContext.RBAC.Rbac().V1().ClusterRoleBinding().OnChange(ctx, handlerName, h.onClusterRoleBinding)
 	rContext.RBAC.Rbac().V1().ClusterRole().OnChange(ctx, handlerName, h.onClusterRole)
 
+	// watch the rio service for changes in order to propagate changes to the ClusterRole controller
+	clusterBindWrap := ClusterScopedRoleBindingWrappper{rContext.RBAC.Rbac().V1().ClusterRoleBinding()}
+	clusterRoleWrap := ClusterScopedRoleWrapper{rContext.RBAC.Rbac().V1().ClusterRole()}
+
+	relatedresource.Watch(ctx, handlerName, resolveToCR, clusterRoleWrap, rContext.Rio.Rio().V1().Service())
+	relatedresource.Watch(ctx, handlerName, resolveToCRBinding, clusterBindWrap, rContext.Rio.Rio().V1().Service())
+
 	return nil
 }
 
+type ClusterScopedRoleWrapper struct {
+	actual rbacv1controller.ClusterRoleController
+}
+
+type ClusterScopedRoleBindingWrappper struct {
+	actual rbacv1controller.ClusterRoleBindingController
+}
+
+func (c ClusterScopedRoleWrapper) Enqueue(namespace string, name string) {
+	c.actual.Enqueue(name)
+}
+
+func (c ClusterScopedRoleBindingWrappper) Enqueue(namespace string, name string) {
+	c.actual.Enqueue(name)
+
+}
+
+// resolver is going to run whenever Rio Service is updated (obj is Rio Service), should only enqueue changes when a service is modified/deleted
+// need to return a list of keys to cluster roles that are associated with that service
+func resolveToCR(namespace, n string, obj runtime.Object) ([]relatedresource.Key, error) {
+	if obj == nil {
+		return nil, nil
+	}
+	meta, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, err
+	}
+	serviceName := meta.GetName()
+	if serviceName == "" {
+		return nil, nil
+	}
+	serviceNamespace := meta.GetNamespace()
+	if serviceNamespace == "" {
+		return nil, nil
+	}
+	// cluster role names follow a format of
+	// rio-$namespace-$serviceName
+	clusterRole := name.SafeConcatName("rio", serviceNamespace, serviceName)
+	return []relatedresource.Key{
+		{
+			Namespace: "",
+			Name:      clusterRole,
+		},
+	}, nil
+}
+
+// resolver is going to run whenever Rio Service is updated (obj is Rio Service), should only enqueue changes when a service is modified/deleted
+// need to return a list of keys to cluster roles that are associated with that service
+func resolveToCRBinding(namespace, n string, obj runtime.Object) ([]relatedresource.Key, error) {
+	if obj == nil {
+		return nil, nil
+	}
+	meta, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, err
+	}
+	serviceName := meta.GetName()
+	if serviceName == "" {
+		return nil, nil
+	}
+	serviceNamespace := meta.GetNamespace()
+	if serviceNamespace == "" {
+		return nil, nil
+	}
```

Comment:
```
service, ok := obj.(*riov1.Service)
```
It might be easy and less code to just assume it is a service object.

Reply:
Added a function to do this